### PR TITLE
Containers vs presentation

### PR DESCRIPTION
**Pull request by ralphcallaway**
@ChuckJonas @DStaleyCCC @DServy 

Container vs Presentation Components
https://callawaycloud.my.salesforce.com/a0K1C00000J3WcX

### CODE REVIEW

**Comment by ChuckJonas** [comment]
@ralphcallaway sorry the instructions weren't super clear.  You should be PR'ing back into your own repo for the skill challenges (not the ChuckJonas/bad-ass-salesforce-stack

File: src/components/counter/counterDisplay.tsx [hunk 1]
```diff
@@ -0,0 +1,28 @@
+import { Progress } from "antd";
+import * as React from "react";
+
+interface ICounterDisplayProps {
+    circleFormat: (percent: number) => string; // no idea wtf this is
```
Comment (ChuckJonas) [inline src/components/counter/counterDisplay.tsx]: `:(percent: number) => string;` just means circle format is a function that takes a number param (percent) and returns a string.   

React composition is always an area of debate.  Passing this function in reduces the cohesiveness of the component, but makes it more flexible.  I typically preferrer cohesiveness over flexibility, until the flexibility is actually needed.

File: src/components/counter/counter.tsx [hunk 2]
```diff
@@ -1,64 +1,62 @@
...
-  }
+    public render() {
+        return (
+            <div>
+                <CounterDisplay
+                    circleFormat={this.formatProgress}
+                    circlePercent={this.state.count / (counterGoal / 100)}
+                    linePercent={(this.state.count % 100)}
+                >
+                    <CounterControls
```
Comment (ChuckJonas) [inline src/components/counter/counter.tsx]: Seems wierd passing this into as a child of DisplayCounter (I would have just had them as siblings), but good demonstration of `props.children`.

File: src/components/counter/counterDisplay.tsx (hunk 1 above)
Comment (ralphcallaway) [reply src/components/counter/counterDisplay.tsx]: did this one a little bit prematurely i guess :) got antsy with all the docs in the typescript deep dive and no coding application

**Pull request closed without merging**